pkg/security/rules/autosuppression/autosuppression.go

```diff
@@ -0,0 +1,116 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package autosuppression holds auto suppression related files
+package autosuppression
+
+import (
+	"strconv"
+	"sync"
+
+	"go.uber.org/atomic"
+
+	"github.com/DataDog/datadog-agent/pkg/security/secl/model"
+	"github.com/DataDog/datadog-agent/pkg/security/secl/rules"
+)
+
+// IsAllowAutosuppressionRule returns true if the given rule allows auto suppression
+func IsAllowAutosuppressionRule(rule *rules.Rule) bool {
+	if val, ok := rule.Definition.GetTag("allow_autosuppression"); ok {
+		b, err := strconv.ParseBool(val)
+		return err == nil && b
+	}
+	return false
+}
+
+// AutoSuppression is an interface for auto suppression
+type AutoSuppression interface {
+	// CanSuppress returns true if the given event can be suppressed
+	CanSuppress(event *model.Event) bool
+	// Apply resets the auto suppression stats based on the given ruleset
+	Apply(ruleSet *rules.RuleSet)
+	// Inc increments the auto suppression stats for the given rule ID
+	Inc(ruleID string)
+	// GetStats returns auto suppressions stats, if enabled
+	GetStats() map[string]int64
+}
```

Comment:
I wonder why did you change the previous struct with each method having a
```
if !s.enabled {
		return nil
}
```
to an interface with a single implementation (+ a stub one)?